sklearn/utils/tests/test_sparsefuncs.py

```diff
@@ -85,6 +85,157 @@ def test_mean_variance_axis1():
             assert_array_almost_equal(X_vars, np.var(X_test, axis=0))
 
 
+@pytest.mark.parametrize(['Xw', 'X', 'weights'],
+                         [
+                         ([[0, 0, 1], [0, 2, 3]],
+                          [[0, 0, 1], [0, 2, 3]],
+                          [1, 1, 1]),
+                         ([[0, 0, 1], [0, 1, 1]],
+                          [[0, 0, 0, 1], [0, 1, 1, 1]],
+                          [1, 2, 1]),
+                         ([[0, 0, 1], [0, 1, 1]],
+                          [[0, 0, 1], [0, 1, 1]],
+                          None),
+                         ([[0, np.nan, 2],
+                           [0, np.nan, np.nan]],
+                          [[0, np.nan, 2],
+                           [0, np.nan, np.nan]],
+                          [1., 1., 1.]),
+                         ([[0, 0],
+                           [1, np.nan],
+                           [2, 0],
+                           [0, 3],
+                           [np.nan, np.nan],
+                           [np.nan, 2]],
+                          [[0, 0, 0],
+                           [1, 1, np.nan],
+                           [2, 2, 0],
+                           [0, 0, 3],
+                           [np.nan, np.nan, np.nan],
+                           [np.nan, np.nan, 2]],
+                          [2., 1.]),
+                         ([[1, 0, 1], [0, 3, 1]],
+                          [[1, 0, 0, 0, 1], [0, 3, 3, 3, 1]],
+                          np.array([1, 3, 1]))
+                         ]
+                         )
+@pytest.mark.parametrize("sparse_constructor",
+                         [sp.csc_matrix, sp.csr_matrix])
+def test_incr_mean_variance_axis_weighted_axis1(Xw, X, weights,
+                                                sparse_constructor):
```

Comment:
Could you please also parametrize with `dtype` in `(np.float64, np.float32)` and check that the dtype of the returned arrays match the input dtype?